Download models with cmake

StuartSulaski opened the pull request.

StuartSulaski committed:
Download models with cmake

saveman71 commented on cmake/Macros.cmake:
```diff
@@ -248,4 +248,41 @@ function(lug_download_thirdparty)
     file(WRITE "${LUG_THIRDPARTY_DIR}/version" "${THIRDPARTY_PLATFORM}_${THIRDPARTY_SHORT_HASH}")
     message(STATUS "Done!")
 
+    file(REMOVE ${DL_FILE})
```
Here you shouldn't remove the downloaded file because it can be reused in a later build (https://github.com/Lugdunum3D/LugBench/pull/29/files#diff-031c6047a0f4c227a313b7b2f3b77219R212)

saveman71 commented on cmake/Macros.cmake:
```diff
@@ -248,4 +248,41 @@ function(lug_download_thirdparty)
+        )
+    endif()
+
+    file(REMOVE "${CMAKE_BINARY_DIR}/${file_name}.zip")
```
Re-use DL_FILE

saveman71 commented on cmake/Macros.cmake:
```diff
@@ -248,4 +248,41 @@ function(lug_download_thirdparty)
+    file(REMOVE "${CMAKE_BINARY_DIR}/${file_name}.zip")
+    
+    message(STATUS "Done!")
+
```
Remove line

saveman71 commented on cmake/Macros.cmake:
```diff
@@ -248,4 +248,41 @@ function(lug_download_thirdparty)
 
+    file(REMOVE ${DL_FILE})
+    file(REMOVE ${DL_FILE_EXPECTED_MD5})
+    
```
Remove line

saveman71 commented on cmake/Macros.cmake:
```diff
@@ -248,4 +248,41 @@ function(lug_download_thirdparty)
+    
+    message(STATUS "Done!")
+
+endfunction()
```
Add newline after end of file (please configure your editor so it does it for you)

saveman71 commented:
You don't call it! And I think there should be some other modifications in this PR too, e.g. removing the old models, changing paths, etc

StuartSulaski committed:
Fixed some things for the pr

StuartSulaski committed:
Improving model download

Models won't be redownloaded if the zip is found
All models will no longer be removed if one fails to unzip
Zip will be stored outside of build/ folder as to not be removed by
mistake

StuartSulaski committed:
Merge branch 'dev' into feature-download-models

StuartSulaski committed:
Update macro so that the zip files are only extracted when there destination directory is missing

StuartSulaski committed:
Removing Duck from resources
